### ISSUE TITLE
Some options not working correctly on Firefox Mobile

### ISSUE THREAD

**Issue by lune-stone** [proginosko/LeechBlockNG#419]
"Filter instead of blocking page" doesn't get applied until opening add-on settings then returning to page.

"Treat referrers as allow-conditions" with block page will always block. With a filter seems to behave correctly except for the issue noted above.

Likely this is just more variations of the same Manifest V3 issues mentioned in other cases.

**Comment by proginosko** [comment]
Yep. All these features depend on a "content script" which Firefox for Android doesn't yet implement properly (see [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1872600)).

Frustrating, isn't it? 🙁

**Comment by lune-stone** [comment]
It's an unfortunate situation to be in for sure.

There is not much that can be done on your end save maybe documenting the list of what is broken somewhere or adding a warning stating something to the effect of "this option uses features that are either experimental or unsupported in your browser" on the configuration page.

Hopefully upstream can reach parity with desktop sooner rather than later. I know this isn't the only add-on that is affected.

**Comment by proginosko** [comment]
Good suggestions.

**Comment by proginosko** [comment]
Update: The bug has been fixed in Firefox Nightly (v128) so hopefully it won't be too long before the fix makes its way to Firefox Beta and then Firefox.
